src/processors/operation-processor.ts

```diff
@@ -36,11 +36,54 @@ export default class OperationProcessor<ResourceT extends Resource> {
 
     if (result !== undefined) {
       eagerLoadedData = await this.eagerLoad(op, result);
+      eagerLoadedData = await this.computeRelationshipProperties(op, eagerLoadedData);
     }
 
     return this.convertToResources(op, result, eagerLoadedData);
   }
 
+  async computeRelationshipProperties(op, eagerLoadedData) {
+    const baseResourceClass = await this.resourceFor(op.ref.type);
+    for (const relationship in eagerLoadedData) {
+      if (eagerLoadedData.hasOwnProperty(relationship)) {
```

Comment:
This `if` is wrapping the whole thing. Let's skip the elements that do not meet this condition as early as possible. Also, we an use the `in` operator for making this statement a bit clearer:

```ts
// If the relationship isn't part of the loaded data,
// move on to the next relationship.
if (!(relationship in eagerLoadedData)) {
  continue;
}
```